sdk/search/search-documents/test/public/utils/recordedClient.ts

```diff
@@ -53,6 +53,15 @@ export const environmentSetup: RecorderEnvironmentSetup = {
 };
 
 export function createClients<IndexModel>(indexName: string): Clients<IndexModel> {
+  switch (testEnv.AZURE_AUTHORITY_HOST) {
```

Comment:
The authority host is not a reliable way to infer the correct scope/endpoint. You can use an Azure Cloud authority host to authenticate into a sovereign cloud resource.

Reply:
@deyaaeldeen Yeah. Do you mean that using `audience` option here instead of the authority host to infer the correct endpoint?

Reply:
@benbp Do you have any ideas about deyaaeldeen's suggestion?

Reply:
@deyaaeldeen and I discussed this offline. We found that the authority host for the public cloud could be used for other clouds for some cognitive services, but that might be a bug and definitely not guaranteed to work with other services. So, you can ignore this comment thread.

For tests, the authority host env variable is set during the ARM template deployment. So, we are good here.